### ISSUE TITLE
Accept HTTP 307 Temporary Redirect status along with 302

### ISSUE THREAD

**Issue by jakublATopera** [websocket-client/websocket-client#952]
https://www.rfc-editor.org/rfc/rfc9110#status.307 (maybe 308 as well).
307 seems like further specification of 302

```
ja@debilian:~/various/websocket-client$ git diff
diff --git a/websocket/_handshake.py b/websocket/_handshake.py
index d28aefd..89f5474 100644
--- a/websocket/_handshake.py
+++ b/websocket/_handshake.py
@@ -32,7 +32,7 @@
 # websocket supported version.
 VERSION = 13
 
-SUPPORTED_REDIRECT_STATUSES = (HTTPStatus.MOVED_PERMANENTLY, HTTPStatus.FOUND, HTTPStatus.SEE_OTHER,)
+SUPPORTED_REDIRECT_STATUSES = (HTTPStatus.MOVED_PERMANENTLY, HTTPStatus.FOUND, HTTPStatus.SEE_OTHER, HTTPStatus.TEMPORARY_REDIRECT)
 SUCCESS_STATUSES = SUPPORTED_REDIRECT_STATUSES + (HTTPStatus.SWITCHING_PROTOCOLS,)
 
 CookieJar = SimpleCookieJar()

```

**Comment by engn33r** [comment]
The link you provided is to the HTTP Semantics RFC, but the relevant text in the [WebSockets hybi-13 spec](https://datatracker.ietf.org/doc/html/draft-ietf-hybi-thewebsocketprotocol-13) that this repository supports is:

> If the status code received from the server is not 101, the client handles the response per HTTP [[RFC2616](https://datatracker.ietf.org/doc/html/rfc2616)] procedures, in particular the client might perform authentication if it receives 401 status code, the server might redirect the client using a 3xx status code (but clients are not required to follow them), etc. Otherwise, proceed as follows.

**Comment by xmuskrat** [comment]
So, this means that you choose to support 301 and 302 redirects, but not 307 redirects because clients can, but are not required, to follow 3xx standards?   Sad, cause I'm hitting this same issue.

**Comment by engn33r** [comment]
The fix was provided in commit https://github.com/websocket-client/websocket-client/commit/69468ad6b5c88c8cec75f83a4b4bd998509ae1ce but a new version has not been made yet. Please use the latest commit for this feature for now.

The text I provided before was clarification because the OP mentioned a RFC that was not the websocket RFC

**Comment by jakublATopera** [comment]
I've had a feeling that since WebSocket RFC release, ws release didn't change, but HTTP protocol WS RFC refers to was changed with addition of mentioned status codes (307, 308), so I just pointed the HTTP change.
Thanks for fix.

**Comment by engn33r** [comment]
@jakublATopera I forgot to push a new release last week, this is now fixed in 1.6.4
